internal/operator/operator.go

```diff
@@ -648,9 +670,10 @@ func (r *minioGroupReconciler) Reconcile(ctx context.Context, req reconcile.Requ
 			return failure(err)
 		}
 
-		l.Info("check if minio group exists")
+		l.Info("get minio group")
 		_, err = mtac.GetGroupDescription(ctx, g.Spec.Name)
-		if err == nil {
+		exists := err == nil
+		if exists && !g.Spec.Migrate {
```

Comment:
Minio groups will _not_ raise an error if they already exist (and instead perform an update).

To support migration workflows, the current existence check is modified to also consider whether `.spec.migrate` is `true`.  If it's true, we proceed with the update.